### PR TITLE
Move LiveFunctionsDataView to DataViews module and add tests

### PR DESCRIPTION
**Pull request by vickyliu-go4it**
We move `LiveFunctionsDataView` to DataViews module and add tests for it.

Tests: Run unit tests
Bugs: http://b/185090791

### CODE REVIEW

**Comment by vickyliu-go4it** [comment]
`LiveFunctionsDataView` calls `app_->GetCaptureData().GetFunctionStatsOrDefault()` to get values for the "Count" and other time statistics columns. I didn't find a solution to directly add fake function stats to a `CaptureData` (I prefer not to add a method to modifying `CaptureData::function_stats_` just for test 🙈 ) so I made some methods needed by `LiveFunctionsDataView` as virtual in order to made a `MockCaptureData`.

**Review by beckerhe — COMMENTED** [review]
Hi Vicky, this looks awesome! Thank you.

The only thing: I'm not sure about `MackCaptureData`. I think we might wanna do that all the way - meaning we should create an abstract base class `CaptureDataInterface` - but that's a lot of work because the type `CaptureData` needs to be replaced by `CaptureDataInterface` in a lot of places.

Another (reasonable)  option would be adding a setter for `FunctionStats` to `CaptureData`. When doing so, is it possible to use `CaptureData` or are there more hurdles? At first sight this sounds simple to me but I'm not sure.

File: src/DataViews/FunctionsDataViewTest.cpp [hunk 1]
```diff
@@ -653,16 +653,22 @@ TEST_F(FunctionsDataViewTest, ContextMenuActionsCallCorrespondingFunctionsInAppI
...
-  EXPECT_CALL(app_, RemoveFrameTrack).Times(1).WillRepeatedly(match_function);
+  EXPECT_CALL(app_, RemoveFrameTrack(testing::An<const orbit_client_protos::FunctionInfo&>()))
+      .Times(1)
+      .WillRepeatedly(match_function);
   view_.OnContextMenu("Unhook", 0, {0});
 
   EXPECT_CALL(app_, SelectFunction).Times(1).WillRepeatedly(match_function);
   EXPECT_CALL(app_, EnableFrameTrack).Times(1).WillRepeatedly(match_function);
-  EXPECT_CALL(app_, AddFrameTrack).Times(1).WillRepeatedly(match_function);
+  EXPECT_CALL(app_, AddFrameTrack(testing::An<const orbit_client_protos::FunctionInfo&>()))
```
[the remark this answers is not in the document]
Comment (vickyliu-go4it) [reply src/DataViews/FunctionsDataViewTest.cpp]: Done. Thanks!

File: src/DataViews/FunctionsDataViewTest.cpp [hunk 2]
```diff
@@ -653,16 +653,22 @@ TEST_F(FunctionsDataViewTest, ContextMenuActionsCallCorrespondingFunctionsInAppI
...
+      .Times(1)
+      .WillRepeatedly(match_function);
   view_.OnContextMenu("Enable frame track(s)", 0, {0});
 
   EXPECT_CALL(app_, DisableFrameTrack).Times(1).WillRepeatedly(match_function);
-  EXPECT_CALL(app_, RemoveFrameTrack).Times(1).WillRepeatedly(match_function);
+  EXPECT_CALL(app_, RemoveFrameTrack(testing::An<const orbit_client_protos::FunctionInfo&>()))
```
[the remark this answers is not in the document]
Comment (vickyliu-go4it) [reply src/DataViews/FunctionsDataViewTest.cpp]: Done. Thanks!

**Comment by vickyliu-go4it** [comment]
> Hi Vicky, this looks awesome! Thank you.
> 
> The only thing: I'm not sure about `MackCaptureData`. I think we might wanna do that all the way - meaning we should create an abstract base class `CaptureDataInterface` - but that's a lot of work because the type `CaptureData` needs to be replaced by `CaptureDataInterface` in a lot of places.
> 
> Another (reasonable) option would be adding a setter for `FunctionStats` to `CaptureData`. When doing so, is it possible to use `CaptureData` or are there more hurdles? At first sight this sounds simple to me but I'm not sure.

Thanks for the suggestion! I changed to adding a `CaptureData::AddFunctionStats` method instead of mocking `CaptureData`.

File: src/DataViews/LiveFunctionsDataViewTest.cpp [hunk 3]
```diff
@@ -0,0 +1,756 @@
...
+    EXPECT_CALL(app_, SetHighlightedFunctionId)
+        .Times(1)
+        .WillOnce([&](uint64_t highlighted_function_id) {
+          EXPECT_EQ(highlighted_function_id, kFunctionIds[2]);
+        });
+
+    view_.OnSelect({2});
+  }
+
+  // Multiple selection
```
Comment (beckerhe) [inline src/DataViews/LiveFunctionsDataViewTest.cpp:644]: Can you document the expected behaviour here? I assume we expect that the first selected function will be set as the highlighted function?

**Review by beckerhe — APPROVED** [review]
Awesome. Thanks Vicky. Two minor Nits. Otherwise LGTM.

File: src/DataViews/LiveFunctionsDataViewTest.cpp [hunk 4]
```diff
@@ -0,0 +1,756 @@
...
+  }
+
+  // Add iterator(s)
+  {
+    const auto add_iterators_index =
+        std::find(context_menu.begin(), context_menu.end(), "Add iterator(s)") -
+        context_menu.begin();
+    ASSERT_LT(add_iterators_index, context_menu.size());
+
+    EXPECT_CALL(live_functions_, AddIterator).Times(1);
```
[the remark this answers is not in the document]
Comment (vickyliu-go4it) [reply src/DataViews/LiveFunctionsDataViewTest.cpp]: Thanks for the suggestions! I added the check for the function id and the function name.

File: src/DataViews/LiveFunctionsDataViewTest.cpp (hunk 3 above)
Comment (vickyliu-go4it) [reply src/DataViews/LiveFunctionsDataViewTest.cpp]: I updated the comments here. But I notice that for the current selection in views, we use `std::set` to remove the duplicate selected indices, but `std::std` will also sort the selected indices. This means that the `LiveFunctionsDataView::OnSelect` always receive sorted rows. I will fix this in another PR.

**Pull request merged**